### PR TITLE
lsp: Correct Path to URI encoding

### PR DESCRIPTION
**Pull request by charlieegan3**
Spaces must be encoded as %20. This case causing the state worker to reload contents from disk as it appeared a file did not exist in the cache.

### CODE REVIEW

File: internal/lsp/server.go [hunk 1]
```diff
@@ -211,18 +211,14 @@ func (l *LanguageServer) StartDiagnosticsWorker(ctx context.Context) {
 		case <-ctx.Done():
 			return
 		case evt := <-l.diagnosticRequestFile:
-			success, err := l.processTextContentUpdate(ctx, evt.URI, evt.Content)
+			err := l.processTextContentUpdate(ctx, evt.URI, evt.Content)
```
Comment (charlieegan3) [inline internal/lsp/server.go:214]: This has been updated to avoid any special handling of parse errors. diagnostics are always sent from the parent now. This is unrelated to the title fix.

File: internal/lsp/server.go [hunk 2]
```diff
@@ -936,33 +932,24 @@ func (l *LanguageServer) processTextContentUpdate(
 	ctx context.Context,
```
Comment (charlieegan3) [inline internal/lsp/server.go:932]: See note above, this function now just attempts to update the parse if the content changed.

File: internal/lsp/server.go [hunk 3]
```diff
@@ -2132,7 +2119,7 @@ func (l *LanguageServer) loadWorkspaceContents(ctx context.Context, newOnly bool
 		}
 
 		// if the caller has requested only new files, then we can exit early
-		if _, ok := l.cache.GetModule(fileURI); newOnly && ok {
+		if _, ok := l.cache.GetFileContents(fileURI); newOnly && ok {
```
Comment (charlieegan3) [inline internal/lsp/server.go:2122]: The best proxy for 'Existence in the cache' is file contents and not a parsed module. This is related to the fix, but is in addition to the required change.

File: internal/lsp/uri/uri.go [hunk 4]
```diff
@@ -19,6 +19,9 @@ func FromPath(client clients.Identifier, path string) string {
 		// Convert Windows path separators to Unix separators
 		path = filepath.ToSlash(path)
 
+		// spaces must be percent-encoded
+		path = strings.ReplaceAll(path, " ", "%20")
```
Comment (charlieegan3) [inline internal/lsp/uri/uri.go:23]: This is the fix, going from paths to URIs when paths contained spaces was broken. This meant files were being reloaded from disk at incorrect times.

Comment (anderseknert) [reply internal/lsp/uri/uri.go:23]: Good find!

**Review by anderseknert — APPROVED** [review]
